deepchecks/nlp/checks/data_integrity/under_annotated_segments.py

```diff
@@ -0,0 +1,290 @@
+# Copyright (C) 2021-2023 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""Module of weak segments performance check."""
+from typing import Dict, List, Tuple, Union
+
+import numpy as np
+import pandas as pd
+import plotly.graph_objects as go
+
+from deepchecks import ConditionCategory, ConditionResult
+from deepchecks.core import CheckResult
+from deepchecks.core.check_result import DisplayMap
+from deepchecks.core.errors import DeepchecksProcessError
+from deepchecks.nlp import Context, SingleDatasetCheck
+from deepchecks.nlp.utils.weak_segments import get_relevant_data_table
+from deepchecks.utils.abstracts.weak_segment_abstract import WeakSegmentAbstract
+from deepchecks.utils.strings import format_number, format_percent
+from deepchecks.utils.typing import Hashable
+
+__all__ = ['UnderAnnotatedMetaDataSegments', 'UnderAnnotatedPropertySegments']
+
+MAX_SAMPLES_IN_FIGURE = 1000
+
+
+class UnderAnnotatedSegments(SingleDatasetCheck, WeakSegmentAbstract):
+    """Check for under annotated data segments."""
+
+    def __init__(self, segment_by: str, columns: Union[Hashable, List[Hashable], None],
+                 ignore_columns: Union[Hashable, List[Hashable], None], n_top_features: int,
+                 segment_minimum_size_ratio: float, n_samples: int,
+                 categorical_aggregation_threshold: float, n_to_show: int, **kwargs):
+        super().__init__(**kwargs)
+        self.segment_by = segment_by
+        self.columns = columns
+        self.ignore_columns = ignore_columns
+        self.n_top_features = n_top_features
+        self.segment_minimum_size_ratio = segment_minimum_size_ratio
+        self.n_samples = n_samples
+        self.n_to_show = n_to_show
+        self.categorical_aggregation_threshold = categorical_aggregation_threshold
+
+    def run_logic(self, context: Context, dataset_kind) -> CheckResult:
+        """Run check."""
+        context.raise_if_token_classification_task(self)
+        context.raise_if_multi_label_task(self)
+
+        text_data = context.get_data_by_kind(dataset_kind)
+        text_data = text_data.sample(self.n_samples, random_state=context.random_state)
+
+        features, cat_features = get_relevant_data_table(text_data, data_type=self.segment_by,
+                                                         columns=self.columns, ignore_columns=self.ignore_columns,
+                                                         n_top_features=self.n_top_features)
+
+        encoded_dataset = self._target_encode_categorical_features_fill_na(features, text_data.label,
```

Comment:
I would think you want to do categorical encoding to the label of isnull(text_data.label), not to the label itself. Would get better partition.